core/domain/interaction_jobs_one_off.py

```diff
@@ -392,3 +393,53 @@ def get_invalid_values(value_type, value, choices):
     @staticmethod
     def reduce(key, values):
         yield (key, values)
+
+
+class LogicProofInteractionOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """Job that produces a list of (exploaration, state) pairs that use the
+    logicProof interaction. This will be used to send manual emails.
+    This job can be deleted after logicProof is removed from the codebase.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [exp_models.ExplorationModel]
+
+    @staticmethod
+    def map(item):
+        if item.deleted:
+            return
+        exploration = exp_fetchers.get_exploration_from_model(item)
+        owner_ids_empty = False
+        found_user_email = False
+        for state in exploration.states.values():
+            if found_user_email:
+                break
+            if state.interaction.id == 'LogicProof':
+                exploration_rights = rights_manager.get_exploration_rights(
+                    item.id, strict=False)
+                if exploration_rights is None:
+                    yield ('MISSING_RIGHTS', 0)
+                    return
+
+                exp_owner_ids = exploration_rights.owner_ids
+                if len(exp_owner_ids) == 0:
+                    owner_ids_empty = True
+                for user_id in exp_owner_ids:
+                    if user_services.get_email_from_user_id(user_id):
+                        found_user_email = True
+                        # Got user email.
+                        break
+            if owner_ids_empty:
+                yield ('EMPTY', 0)
+            owner_ids_empty = False # For the next state.
+
+        if found_user_email:
+            yield ('SUCCESS', 1)
```

Comment:
No, in the case when we got email, yield email together with the exp ID. In all cases (any exploration) yield `SUCCESS`.